test/regress/run_regression.py

```diff
@@ -24,8 +25,30 @@
 COMMAND_LINE = 'COMMAND-LINE: '
 
 
+def run_process(args, cwd, timeout, s_input=None):
+    """Runs a process with a timeout `timeout` in seconds. `args` are the
+    arguments to execute, `cwd` is the working directory and `s_input` is the
+    input to be sent to the process over stdin."""
+
+    proc = subprocess.Popen(
+        args,
+        cwd=cwd,
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+
+    timer = threading.Timer(timeout, lambda p: p.kill(), [proc])
+    try:
+        timer.start()
+        out, err = proc.communicate(input=s_input)
+    finally:
+        timer.cancel()
```

Comment:
You might want to indicate that an error status has occurred on time outs. (Maybe this is already taken care of. In which case, you might want to update the documentation.)

Reply:
That's a good point. I changed the code to explicitly initialize the variables and return 124 if the process times out (that seems to be the return code when you use the `timeout` util in Linux.